### ISSUE TITLE
Upgrade to Orchard Core 3.0 and .NET 10 (OSOE-925)

### ISSUE THREAD

**Issue by Piedone** [Lombiq/Open-Source-Orchard-Core-Extensions#894]
...once it's released.

- [x] Review [OC 3.0 release notes](https://github.com/OrchardCMS/OrchardCore/blob/main/src/docs/releases/3.0.0.md) and make necessary changes.
- [x] Keep https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-9/ in mind. In particular, use `System.Threading.Lock` where appropriate. Also see https://maoni0.medium.com/preparing-for-the-net-10-gc-88718b261ef2 (e.g., DATAS is now enabled by default).
- [x] Swagger is now gone so the features using it [in the UI Testing Toolbox](https://github.com/search?q=repo%3ALombiq%2FUI-Testing-Toolbox%20swagger&type=code) need to be adapted. Also see [this post](https://codewithmukesh.com/blog/dotnet-swagger-alternatives-openapi).
- [x] Change all sync-over-async site settings accesses [like this](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/160e1bd8d1eecb9cbb88d4650beb8195f33557dd/Lombiq.TrainingDemo/Services/DemoSettingsConfiguration.cs#L41) to use the new sync `GetSiteSettings` method from https://github.com/OrchardCMS/OrchardCore/pull/17798.
- [x] In UITT mark `Lombiq.Tests.UI.Helpers.AppLogAssertionHelper.NotMediaCacheEntries` as `Obsolete` and make sure it's not used. It's no longer needed since https://github.com/OrchardCMS/OrchardCore/pull/18341 was merged.
- [x] Remove `System.Linq.Async` package references, since that's now part of .NET.
- [x] Revert this workaround: https://github.com/Lombiq/Hosting-Media-Theme/pull/98/commits/a5861d7559ef452d42136f0fd7cc609ede03afb8.
- [x] Upgrade Scrutor to the latest version and check if https://github.com/khellang/Scrutor/issues/268 doesn't happen due to .NET 10. If it doesn't, then also remove [it being disabled in Renovate](https://github.com/Lombiq/renovate-config/blob/fdbd994e93c3f8f9057555d08fb2c427dbd05981/default-dotnet.json5#L51-L56).
- [x] Check if https://github.com/Lombiq/Hosting-Tenants/issues/161 is fixed.
- [x] Upgrade `Moq.AutoMock` to latest and remove [it being disabled in Renovate](https://github.com/Lombiq/renovate-config/blob/2563c576caaf8a89fbd86f588a63c13a18c2ef86/default-dotnet.json5#L57-L66).
- [x] Fix breaking changes in Lombiq.Walkthroughs
- [x] Fix breaking changes with Elasticsearch integration.
- [x] Fix Elasticsearch not working in CI only (it's fine in local).

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-925)

**Comment by sarahelsaig** [comment]
I _think_ https://github.com/Lombiq/Hosting-Tenants/issues/161 is fixed, but I'm not 100% certain. Since reproduction is supposedly more reliable on DotNest, I've added a note to check it during NEST-608 as well.

**Comment by Piedone** [comment]
OK, thanks!

**Comment by Piedone** [comment]
"Fix Elasticsearch not working in CI only (it's fine in local)." is done, right?

**Comment by Piedone** [comment]
Not fixed yet, only after merging https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/1211.

**Comment by lxkn** [comment]
Has work on this task been suspended?

**Comment by sarahelsaig** [comment]
We have done everything on our end, but we will only merge the associated pull request when Orchard Core 3.0 is released. You can track their release status in this issue: https://github.com/OrchardCMS/OrchardCore/issues/18705

You could also use our alpha NuGet packages (the ones with the `-alpha.N.osoe-925` suffix where N is a number, such as [Lombiq.HelpfulLibraries@12.6.1-alpha.7.osoe-925](https://www.nuget.org/packages/Lombiq.HelpfulLibraries/12.6.1-alpha.7.osoe-925)) because they are good to go according to our internal testing, if you are already using the OC preview packages [from Cloudsmith](https://cloudsmith.io/~orchardcore/repos/preview/packages/). But I wouldn't recommend that in production.

**Comment by Piedone** [comment]
It would be great to merge https://github.com/Lombiq/UI-Testing-Toolbox/pull/770 before this, because that (due to Selenium BiDi changes) contains breaking changes too. However, that waits for https://github.com/dequelabs/axe-core-nuget/issues/252.

**Comment by sarahelsaig** [comment]
I'm not sure if adding an inactive blocker (last comment has been over a month ago) is wise.

**Comment by Piedone** [comment]
Yeah, unless something like the maintainers saying on Monday that there will be a release that week.

**Comment by Piedone** [comment]
And if not, I'll add a custom class for browser log entries so while we'll still have a breaking change now, not in the future (also because this, what's breaking now, was broken at least twice before too).
